commit bc74b0c8af17458ecae77f725e507ab5fd100105
Author: Martin Willi <martin@strongswan.org>
Date:   Wed Nov 25 00:58:39 2009 +0000

    xfrm: Add SHA384 and SHA512 HMAC authentication algorithms to XFRM
    
    These algorithms use a truncation of 192/256 bits, as specified
    in RFC4868.
    
    Signed-off-by: Martin Willi <martin@strongswan.org>
    Acked-by: Herbert Xu <herbert@gondor.apana.org.au>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/xfrm/xfrm_algo.c b/net/xfrm/xfrm_algo.c
index b39341072aa6..ef8d61d50462 100644
--- a/net/xfrm/xfrm_algo.c
+++ b/net/xfrm/xfrm_algo.c
@@ -199,6 +199,40 @@ static struct xfrm_algo_desc aalg_list[] = {
 		.sadb_alg_maxbits = 256
 	}
 },
+{
+	.name = "hmac(sha384)",
+
+	.uinfo = {
+		.auth = {
+			.icv_truncbits = 192,
+			.icv_fullbits = 384,
+		}
+	},
+
+	.desc = {
+		.sadb_alg_id = SADB_X_AALG_SHA2_384HMAC,
+		.sadb_alg_ivlen = 0,
+		.sadb_alg_minbits = 384,
+		.sadb_alg_maxbits = 384
+	}
+},
+{
+	.name = "hmac(sha512)",
+
+	.uinfo = {
+		.auth = {
+			.icv_truncbits = 256,
+			.icv_fullbits = 512,
+		}
+	},
+
+	.desc = {
+		.sadb_alg_id = SADB_X_AALG_SHA2_512HMAC,
+		.sadb_alg_ivlen = 0,
+		.sadb_alg_minbits = 512,
+		.sadb_alg_maxbits = 512
+	}
+},
 {
 	.name = "hmac(rmd160)",
 	.compat = "rmd160",